yb-voyager/src/srcdb/yugbaytedb_test.go

```diff
@@ -59,6 +61,193 @@ func TestYugabyteGetAllTableNames(t *testing.T) {
+
+		//-- 1. SERIAL column (implicit)
+		`CREATE TABLE public.serial_table (
+			id SERIAL PRIMARY KEY,
+			name TEXT
+		);`,
+
+		//-- 2. BIGSERIAL column
+		`CREATE TABLE public.bigserial_table (
+			id BIGSERIAL PRIMARY KEY,
+			description TEXT
+		);`,
+
+		//-- 3. IDENTITY columns (Generated Always)
+		`CREATE TABLE public.identity_always_table (
+			id BIGINT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+			info TEXT
+		);`,
+
+		//-- 4. IDENTITY columns (Generated By Default)
+		`CREATE TABLE public.identity_default_table (
+			id BIGINT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+			comments TEXT
+		);`,
+
+		//-- 5. DEFAULT nextval() with a manually created seq
+		`CREATE SEQUENCE public.manual_seq START 1000;`,
+
+		`CREATE TABLE public.default_nextval_table (
+			id BIGINT NOT NULL DEFAULT nextval('public.manual_seq'),
+			value TEXT
+		);`,
+
+		//-- 6. Same as above but cross-schema
+		`CREATE SEQUENCE custom_schema.cross_schema_seq;`,
+
+		`CREATE TABLE public.cross_schema_default_seq_table (
+			id BIGINT NOT NULL DEFAULT nextval('custom_schema.cross_schema_seq'),
+			payload TEXT
+		);`,
+
+		//-- 7. A table with a manually owned by sequence (DEFAULT)
+		`CREATE SEQUENCE public.manual_linked_seq;`,
+
+		`CREATE TABLE public.manual_linked_table (
+			id BIGINT NOT NULL,
+			meta TEXT
+		);`,
+		`ALTER TABLE public.manual_linked_table ALTER COLUMN id SET DEFAULT nextval('public.manual_linked_seq');`,
+		`ALTER SEQUENCE public.manual_linked_seq OWNED BY  public.manual_linked_table.id;`,
+
+		//--8 table with manually owned sequence on text column
+		`CREATE SEQUENCE custom_schema.user_code_seq
+		START WITH 1
+		INCREMENT BY 1;`,
+
+		`CREATE OR REPLACE FUNCTION generate_user_code() RETURNS TEXT AS $$
+		DECLARE
+			new_code TEXT;
+		BEGIN
+			SELECT 'USR' || LPAD(nextval('custom_schema.user_code_seq')::TEXT, 4, '0') 
+			INTO new_code;
+			RETURN new_code;
+		END;
+		$$ LANGUAGE plpgsql;
+
+		CREATE TABLE custom_schema.users (
+			user_id int PRIMARY KEY,
+			user_code TEXT UNIQUE,
+			name TEXT
+		);`,
+
+		`ALTER TABLE ONLY custom_schema.users ALTER COLUMN user_code SET DEFAULT generate_user_code();`,
+
+		`ALTER SEQUENCE custom_schema.user_code_seq OWNED BY custom_schema.users.user_code;`,
+
+		//--9 normal sequences not attached to tables
+		`CREATE SEQUENCE seq1_unattached;`,
+		`CREATE SEQUENCE custom_schema.seq2_unattached;`,
+
+		//--10 manually linked sequence to multiple columns
+		`CREATE SEQUENCE public.manual_linked_seq_another ;`,
+
+		`CREATE TABLE public.manual_linked_table_1 (
+			id BIGINT NOT NULL,
+			meta TEXT
+		);`,
+		`ALTER TABLE public.manual_linked_table_1 ALTER COLUMN id SET DEFAULT nextval('public.manual_linked_seq_another');`,
+
+		`CREATE TABLE public.manual_linked_table_2 (
+			id BIGINT NOT NULL,
+			meta TEXT
+		);`,
+		`ALTER TABLE public.manual_linked_table_2 ALTER COLUMN id SET DEFAULT nextval('public.manual_linked_seq_another');`,
+	)
+	defer testYugabyteDBSource.TestContainer.ExecuteSqls(
```

Comment:
in general, i wonder if dropping the entire database will make it easier for clean up functions. (don't have to worry about DROPPING every object)

Reply:
yeah should be possible, will try to do this as whole separately for all integration tests we have

Reply:
yeah thats a good idea.
Will have to handle the update of db name in container config and ability to use custom dbname for each test case.